hack/verify-golangci-lint.sh

```diff
@@ -18,20 +18,12 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
-VERSION=v1.42.0
-URL_BASE=https://raw.githubusercontent.com/golangci/golangci-lint
-URL=$URL_BASE/$VERSION/install.sh
+# version v1.42.0
+IMAGE_SHA=sha256:8d0a2ad834fe92f69fe69918a68b0c141f0ef97980e6bba84b72bdd7d58fc3b5
 
 if [[ ! -f .golangci.yml ]]; then
     echo 'ERROR: missing .golangci.yml in repo root' >&2
     exit 1
 fi
 
-if ! command -v golangci-lint; then
-    curl -sfL $URL | sh -s $VERSION
-    PATH=$PATH:bin
-fi
-
-golangci-lint version
-golangci-lint linters
-golangci-lint run "$@"
+docker run --rm -v $(pwd):/app -w /app golangci/golangci-lint@$IMAGE_SHA golangci-lint run "$@"
```

Comment:
if we decide to move with this approach we will need to change the prow job and make sure we have a docker running

the current job fails for this

```
./hack/verify-golangci-lint.sh: line 29: docker: command not found
```

Reply:
sounds good. Im happy to make those changes if you can point me to the location where the prow config is.

Reply:
I switched to just validating the checksum of the golangci-lint install script. PTAL. Thanks.